rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -562,6 +572,63 @@ class Client : public ClientBase
     return async_cancel(cancel_request, cancel_callback);
   }
 
+  /// Stops the callbacks for the goal in a thread safe way
+  /**
+   * This will NOT cancel the goal, it will only stop the callbacks.
+   *
+   * After the call to this function, it is guaranteed that there
+   * will be no more callbacks from the goal. This is not guaranteed
+   * if multiple threads are involved, and the goal_handle is just
```

Comment:
this `This is not guaranteed if multiple threads are involved` doc section, why is it not guaranteed if multi-threaded are involved?

Reply:
Race conditions

Reply:
If you drop the handle without calling this function you have no guarantee that no further callback will happen.